src/inmanta/resources.py

```diff
@@ -400,16 +400,15 @@ def object_to_id(
 
                 agent_value = getattr(agent_value, el)
 
-            except inmanta.ast.UnsetException as e:
-                raise e
-            except inmanta.ast.UnknownException as e:
-                raise e
-            except inmanta.ast.UnexpectedReference:
-                current_path: str = ".".join(path_elements[: i + 1])
-                raise ResourceException(
-                    "Encountered reference in resource's agent attribute. Agent attribute values can not be references."
-                    f" Encountered at attribute {current_path!r} of resource instance {model_object}"
-                )
+                if isinstance(agent_value, references.Reference):
```

Comment:
bit of a special case, we may actually want to not allow any reference, but that can be done later.